tests/containers/podman_netavark.pm

```diff
@@ -0,0 +1,147 @@
+use containers::utils qw(get_podman_version registry_url);
+use transactional qw(trup_call check_reboot_changes);
+use utils qw(zypper_call);
+use Utils::Systemd qw(systemctl);
+
+sub remove_subtest_setup {
+    assert_script_run("podman container rm -af");
+    assert_script_run("podman network prune -f");
+    assert_script_run("podman network ls");
+    assert_script_run("podman ps -a");
+}
+
+sub _cleanup {
+    my $podman = shift->containers_factory('podman');
+    select_console 'log-console';
+    remove_subtest_setup;
+    script_run('rm -rf /etc/containers/containers.conf');
+    $podman->cleanup_system_host();
+    validate_script_output('podman info --format {{.Host.NetworkBackend}}', sub { /cni/ });
+    validate_script_output('podman network ls', sub { /podman\s+bridge/ });
+}
+
+sub switch_to_netavark {
+    my @pkgs = qw(netavark aardvark-dns);
+
+    if (is_transactional) {
+        trup_call("pkg install @pkgs");
+        check_reboot_changes;
+    } else {
+        zypper_call("in @pkgs");
+    }
+
+    # change network backend to *netavark*
+    assert_script_run(q(echo -e '[Network]\nnetwork_backend="netavark"' >> /etc/containers/containers.conf));
+    # reset the storage back to the initial state
+    assert_script_run('podman system reset');
+    validate_script_output('podman info --format {{.Host.NetworkBackend}}', sub { /netavark/ });
+}
+
+sub run {
+    my ($self, $args) = @_;
+
+    select_serial_terminal;
+    my $podman = $self->containers_factory('podman');
+    my $podman_version = get_podman_version();
+
+    if (package_version_cmp($podman_version, '4.4.2') < 0) {
+        record_info('No support', "Netavark backend is not supported in podman-$podman_version");
+        return 1;
+    }
+
+    switch_to_netavark;
+    $podman->cleanup_system_host();
+
+    ## TEST1
+    record_info('TEST1', 'set static IP, and MAC addresses on a per-network basis');
+    my $testnet = 'test_simple';
+    my $test_ctr = get_required_var('REGISTRY') . '/fedora/apache';
+
+    assert_script_run("podman network create --gateway 10.90.0.1 --subnet 10.90.0.0/16 $testnet");
+    assert_script_run("podman run --network $testnet:ip=10.90.0.8,mac=76:22:33:44:55:66 -d --name test_apache $test_ctr");
+    assert_script_run("podman container inspect test_apache --format {{.NetworkSettings.Networks.$testnet.IPAddress}}");
+    validate_script_output("curl -I -s 10.90.0.8:80", sub { /HTTP.* 200 OK/ });
+    assert_script_run("grep test_apache /run/containers/networks/aardvark-dns/$testnet");
+    remove_subtest_setup;
+
+    ## TEST2
+    record_info('TEST2', 'set static IPs for containers and check their networking');
+    $testnet = 'primary';
+    my $testnet2 = 'secondary';
+    my $test_ctr_sec = 'registry.opensuse.org/bci/bci-busybox';
+
+    assert_script_run("podman network create --gateway 10.90.0.1 --subnet 10.90.0.0/16 $testnet");
+    assert_script_run("podman network create --gateway 10.64.0.1 --subnet 10.64.0.0/16 $testnet2");
```

Comment:
nit: moving the IPs out to variables might also make this more readable and error-prone, hopefully